internal/protocoltest/awsrestjson/internal/endpoints/endpoints.go

```diff
@@ -0,0 +1,92 @@
+// Code generated by smithy-go-codegen DO NOT EDIT.
+package endpoints
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws"
+	endpoints "github.com/aws/aws-sdk-go-v2/aws/endpoints/v2"
+	"regexp"
+)
+
+type EndpointResolver struct {
+	options    endpoints.ResolveOptions
+	partitions endpoints.Partitions
+}
+
+func (r *EndpointResolver) ResolveEndpoint(region string) (aws.Endpoint, error) {
+	return r.partitions.EndpointFor(region, r.options)
+}
+
+func NewResolver(o endpoints.ResolveOptions) *EndpointResolver {
+	return &EndpointResolver{
+		options:    o,
+		partitions: DefaultPartitions,
+	}
+}
+
+var DefaultPartitions = endpoints.Partitions{
+	{
+		ID: "aws",
+		Defaults: endpoints.Endpoint{
+			Hostname:          "restjson.{region}.amazonaws.com",
+			Protocols:         []string{"https"},
+			SignatureVersions: []string{"v4"},
+		},
+		RegionRegex: func() *regexp.Regexp {
+			r, _ := regexp.Compile("^(us|eu|ap|sa|ca|me|af)\\-\\w+\\-\\d+$")
```

Comment:
Maybe this should be a must compile to capture regex modeling errors during CI/CD? The must compile would also get rid of the extra inline function call.

Reply:
Yeah I was debating on this, it would require that we attempted to execute the code to ensure that the `MustCompile` checks pass.